src/Microsoft.Health.Fhir.Azure/ExportDestinationClient/AzureExportDestinationClient.cs

```diff
@@ -104,24 +107,57 @@ public async Task CommitAsync(CancellationToken cancellationToken)
                 CloudBlockBlobWrapper blobWrapper = _uriToBlobMapping[mapping.Key.Item1];
 
                 var blockId = Convert.ToBase64String(Encoding.ASCII.GetBytes(mapping.Key.Item2.ToString("d6")));
-                uploadAndCommitTasks.Add(Task.Run(async () =>
-                {
-                    await blobWrapper.UploadBlockAsync(blockId, stream, md5Hash: null, cancellationToken);
-                    await blobWrapper.CommitBlockListAsync(cancellationToken);
 
-                    stream.Dispose();
-                }));
+                uploadTasks.Add(blobWrapper.UploadBlockAsync(blockId, stream, md5Hash: null, cancellationToken));
+                wrappersToCommit.Add(blobWrapper);
+            }
+
+            await Task.WhenAll(uploadTasks);
+
+            // Commit all the blobs that were uploaded.
+            var commitTasks = new List<Task>();
+            foreach (CloudBlockBlobWrapper wrapper in wrappersToCommit)
+            {
+                commitTasks.Add(wrapper.CommitBlockListAsync(cancellationToken));
             }
 
-            await Task.WhenAll(uploadAndCommitTasks);
+            await Task.WhenAll(commitTasks);
 
             // We can clear the stream mappings once we commit everything in memory.
+            foreach (Stream stream in _streamMappings.Values)
+            {
+                stream.Dispose();
+            }
+
             _streamMappings.Clear();
         }
 
+        public async Task InitializeForResumingExportAsync(CancellationToken cancellationToken)
```

Comment:
Instead of downloading all blobs, can we create a method like OpenFileAsync and open individual one as needed since we know the file name? Or make the method CreateIfNotExistAsync or something?

Reply:
Renamed the method to OpenFilesAsync and passed in the list of specific files that it needs to open for resuming export.